clustering/web/cache/src/main/java/org/wildfly/clustering/web/cache/session/SessionCreationMetaData.java

```diff
@@ -28,11 +28,31 @@
  * The more static aspects of a session's meta-data.
  * @author Paul Ferraro
  */
-public interface SessionCreationMetaData extends ImmutableSessionCreationMetaData {
-
+public interface SessionCreationMetaData extends ImmutableSessionCreationMetaData, AutoCloseable {
+    /**
+     * Sets the maximum duration of time this session may remain idle before it will be expired by the session manager.
+     * @param a maximum duration of time this session may remain idle before it will be expired by the session manager.
+     */
     void setMaxInactiveInterval(Duration duration);
 
+    /**
+     * Indicates whether or not this session has been invalidated.
+     * @return true, if this session was invalidated, false otherwise.
+     */
     boolean isValid();
 
+    /**
+     * Invalidates this session.
+     * @return true, if this session was previous valid, false otherwise
+     */
     boolean invalidate();
+
+    /**
+     * Indicates whether or not this session was newly created.
+     * @return true, if this session was newly created, false otherwise.
+     */
+    boolean isNew();
+
+    @Override
+    void close();
```

Comment:
What is the expected use of this? From looking at the PR it seems it's meant to be called at some point and that changes what isNew() would return. But 'close()' as a method name doesn't say that, which is ok so long as the expected use of close() will have that effect. But the expected use is ???.

Reply:
This is ultimately triggered by org.wildfly.clustering.web.session.Session#close which effectively ends the transient lifecycle of the session (typically spanning a single request).

Reply:
@pferraro Thanks. I meant to ask in my previous comment whether the javadoc should say something about this, for the benefit of future maintainers.

Reply:
More javadoc cannot hurt.

Reply:
Thanks. That helps a lot. I think of a session as a long lived thing. Object implementing these interfaces might or might not be, so my brain had trouble grokking what the semantics of close would be.